### PR TITLE
code rev for bill

### PR DESCRIPTION
**Pull request by tolgaek**

### CODE REVIEW

File: app/routes/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+var changeCase = require('change-case');
+var fs = require('fs');
+
+module.exports = function(app, Router) {
+  var routes = fs.readdirSync('./');
+  //Sanitize file names
```
Comment (wcurtis) [inline app/routes/index.js]: Personal preference, but I'd add a space after `//` before the comment text.

Either way, be consistent (below in this file you do include spaces in some comments)

Comment (tolgaek) [inline app/routes/index.js]: good point, thanks, I actually like with space as well. updating

File: app/routes/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  routes = routes.map(function(routeName) {
+    return routeName.replace(/\.[^/.]+$/, "");
+  });
+  
+  //Initialize all routes
+  routes.forEach(function(routeName) {
+    if(routeName !== "index") {
```
Comment (wcurtis) [inline app/routes/index.js]: Personal preference again, but I like adding a space between `if` and `(`.

Nice job on the triple equals check

File: app/routes/users.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      res.json(user);
+    })
+    .catch(function(err) {
+      return next(err);
+    });
+  }) 
+  .put(function(req, res, next) {
+    User({ u_id: req.params.userId, u_username: 'tolga'}, {saved:true}).save({force: true})
+    .then(function() { 
+      return res.sendStatus(200)
```
Comment (wcurtis) [inline app/routes/users.js]: Missing `;` at the end of this line

File: app/routes/users.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    //Get USER by id and return as json
+    User.find(req.params.userId).then(function(user) {
+      res.json(user);
+    })
+    .catch(function(err) {
+      return next(err);
+    });
+  }) 
+  .put(function(req, res, next) {
+    User({ u_id: req.params.userId, u_username: 'tolga'}, {saved:true}).save({force: true})
```
Comment (wcurtis) [inline app/routes/users.js]: Should `tolga` be hard coded?

Comment (tolgaek) [inline app/routes/users.js]: oh, nice catch, I'm uploading a better user route as example

File: config/initializers/database.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+// config/initializers/database.js
+
+var config = require("nconf");
+
+module.exports = function(cb) {
+  //Initialize the component here then call the callback
```
Comment (wcurtis) [inline config/initializers/database.js]: Style again, use block commenting for multi-line stuff

```javascript
/**
 * Detailed comments
 * like this
 */
```

File: config/initializers/server.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+  //Initialize Authentication
+  require('../../app/auth/auth')(app);
+
+  logger.info('[SERVER] Initializing routes');
+  require('../../app/routes/index')(app, express.Router);
+
+  app.use(express.static(path.join(__dirname, 'public')));
+
+  if (app.get('env') === 'development') {
```
Comment (wcurtis) [inline config/initializers/server.js]: This is pretty cool

File: index.js [hunk 7]
```diff
@@ -25,25 +16,19 @@ nconf.env();
...
-    },
-    function initializeDBConnection(redis, wfCB) {
-      GLOBAL.redis = redis
-      require('./config/initializers/databaseConnection')(wfCB)
-    },
-    function initializeModels(db, wfCB) {
-      GLOBAL.db = db;
-      wfCB();
+async.series([
+    function initializeDBConnection(callback) {
```
Comment (wcurtis) [inline index.js]: Should be a 2 space indent here, not 4

File: app/models/user.js [hunk 8]
```diff
@@ -0,0 +1,5 @@
+//app/models/user.js
```
Comment (wcurtis) [inline app/models/user.js]: For simplicity, I'd delete this file if it's not being used

File: app/routes/users.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+      return res.sendStatus(200)
+    })
+    .catch(function(e) {
+      logger.error('Something went wrong', e);
+      return next(e);
+    });
+  });
+
+  router.route('/')
+  .get(function(req, res, next) {
```
Comment (wcurtis) [inline app/routes/users.js]: What about `patch/put/delete`? May as well stub those out too so I know they're avaialable

**Comment by wcurtis** [comment]
Ok that's about all I got. Very nice overall :+1:

File: app/routes/users.js (hunk 9 above)
Comment (tolgaek) [inline app/routes/users.js]: good point, I will add them quickly

File: app/models/user.js (hunk 8 above)
Comment (tolgaek) [inline app/models/user.js]: i'm adding a stub that just returns JSON as dummy

File: app/routes/index.js (hunk 1 above)
Comment (tolgaek) [inline app/routes/index.js]: fixed

**Comment by wcurtis** [comment]
Wohoo! :ship:

**Pull request closed without merging**